src/sources/socket/mod.rs

```diff
@@ -34,7 +34,11 @@ pub enum Mode {
 }
 
 impl SocketConfig {
-    pub fn make_tcp_config(addr: SocketAddr) -> Self {
+    pub fn make_tcp_config(tcp_config: tcp::TcpConfig) -> Self {
```

Comment:
Are these only used for tests? Might be able to put them behind a `#[cfg(test)]`

Reply:
I liked your suggestion and tried, but it doesn't work for two reasons:
1) it's used in benches
2) it's used in integration tests which build the module without the test config